### PR TITLE
[WIP] Auto-Generate and publish winery update-site 

### PR DESCRIPTION
**Pull request by Vogel612**
OpenTOSCA/container wants to use winery modules. Especially for development, SNAPSHOTs of said modules need to be available.

For a low cycle-time it is thus imperative to have control over the publishing of SNAPSHOTs to an OSGI-queryable site. To automate this, the plugin `p2-maven-plugin` is introduced and configured to provide a subset of eclipse winery components as a github-hosted p2 repository.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -44,8 +44,27 @@ deploy:
       all_branches: true
       condition: $AWSAccessKeyId
   - provider: script
-    script: "mvn com.github.github:site-maven-plugin:site -N"
+    script: "mvn org.reficio:p2-maven-plugin:site -N"
     skip_cleanup: true
     on:
       branch: ustutt
-      tags: true
```
Comment (koppor) [inline .travis.yml]: No tags any more?

File: pom.xml [hunk 2]
```diff
@@ -198,16 +198,104 @@
...
                         <execution>
-                            <goals>
-                                <goal>site</goal>
-                            </goals>
-                            <phase>site</phase>
+                            <id>default-cli</id>
+                            <configuration>
+                                <artifacts>
+                                    <artifact>
+                                        <id>org.eclipse.winery:org.eclipse.winery.model.bpmn4tosca:2.0.0-SNAPSHOT</id>
```
Comment (koppor) [inline pom.xml]: We really need to give specific versions here? So at each release, we need to update this file?

Comment (miwurster) [reply pom.xml]: Yes, according to the docs it seems so.

File: .travis.yml (hunk 1 above)
Comment (Vogel612) [reply .travis.yml]: the conditions might be cumulative, meaning both would need to apply, which is not intended.

Comment (miwurster) [reply .travis.yml]: Yeah, indeed. We changed it to `all_branches`...

**Pull request merged**